### ISSUE TITLE
WARNING: couldn't find transcript named [XXXXXX]; returning transcript  as it's own gene

### ISSUE THREAD

**Issue by gresteban** [COMBINE-lab/salmon#98]
Hi Robb,

I'm getting the following warning message in all my runs of Salmon (0.7.2). Always the same transcript for all the libraries. The transcript reported has, indeed, no gene associated in the genemap file but it appears in the "quant.gene.sf" output file.  I've checked with a different dataset (different species, libraries and transcriptome reference) and there's always also one transcript reported. I'm feeding Salmon with the genemap in the 2-columns format. Do you have any clue about what could be happening?

Thanks!

```
Computing gene-level abundance estimates
There were 25081 transcripts mapping to 11607 genes
Parsed 54000 expression linesWARNING: couldn't find transcript named [v3_99998_1_1]; returning transcript  as it's own gene

done
Aggregating expressions to gene level . . . done
```

**Comment by rob-p** [comment]
Hi @gresteban,

  Yes, here's what's happening.  Salmon's aggregation strategy is to collect the transcript-level abundances and then aggregate them to the gene level (this means summing the counts / TPMs, and doing an abundance weighted combination of the lengths / effective lengths).  If there is no gene annotated for a transcript, than Salmon's behavior is to treat this transcript as it's own (single isoform) gene, and the gene name should be the name of this transcript.  This is to retain the totality of the originally computed expression (i.e. the gene level results will have the same total read count and TPM as the transcript level results).  I realize that this default behavior may not be a one-size-fits-all solution.  If you have a suggestion for a different way to handle this, I'd be happy to hear it.  I'll double check the code, but I believe the reason you're only seeing the warning show up once is that Salmon reports the warning the first time it sees such an occurrence, but not thereafter.  That's sort of confusing default behavior, so I'll fix that up anyway.

  Also, I should mention that, generally, I now recommend that people use [tximport](https://bioconductor.org/packages/release/bioc/html/tximport.html) for transcript to gene level aggregation.  Within a single sample, it's default strategy is equivalent to what Salmon does.  However, it has some nice extra features, like the ability to do transcript -> gene aggregation across multiple samples while accounting for the variation in the effective gene length (abundance weighted average of isoform lengths) across samples.  Let me know if you have other questions about this.

**Comment by gresteban** [comment]
Hi @rob-p,

Thanks for your responsiveness and detailed answer as usual.

I was mostly concerned for seeing it happening only once when there are many more transcripts with no gene associated. Since reporting a warning for each transcript with no gene associated could be too noisy, maybe reporting a warning with the total number of transcripts without annotation would be enough. (And not placing any transcript in the "quant.gene.sf" file).

Btw, thanks for the possibility to generate a .sam file introduced in version 0.7.2. That's very useful for some of the analyses I'll have to perform with my data. I appreciate that.

**Comment by rob-p** [comment]
The underlying bug that was causing this behavior (as well as improvements in reporting) have been fixed in v0.8.0.

**Comment by CuriusScientist** [comment]
@rob-p I am new in the field on RNA-Seq analysis and trying to understand the methodology behind Salmon's aggregation strategy. 
can you explain it in detail for this example which I ended up for a test analysis? Here I have shown the transcripts and the resulting gene.

Name	Length	EffectiveLength	TPM	NumReads
ENSG00000172530	78.6693	60.7441	9.11087	516.593
				
ENST00000569400	802	553	0.0341855	1.27033
ENST00000423252	644	395	0.0573103	1.52117
ENST00000488074	759	510	0	0
ENST00000439677	736	487	1.10087	36.0257
ENST00000454563	558	309	0	0
ENST00000479780	1551	1302	0.382663	33.4793
ENST00000393208	2283	2034	0.374088	51.1297
ENST00000459966	627	378	0	0
ENST00000412691	586	337	1.58787	35.9578
ENST00000526460	547	298	0	0
